autoscale-api/src/main/java/com/sequenceiq/periscope/api/endpoint/v1/DistroXAutoScaleScalingActivityV1Endpoint.java

```diff
@@ -0,0 +1,147 @@
+import io.swagger.annotations.ApiOperation;
+
+@Validated
+@Path("/v1/scaling_activities")
+@Api(value = "/v1/scaling_activities", description = SCALING_ACTIVITY_DESCRIPTION, protocols = "http,https")
+public interface DistroXAutoScaleScalingActivityV1Endpoint {
+
+    /**
+     * Returns Scaling activities in a particular duration.
+     * @param clusterName Name of cluster for which we require the scaling activities.
+     * @param durationInMinutes the duration in which we want to get all the scaling activities.
+     * @return List of scaling Activities in the duration.
+     */
+    @GET
+    @Path("cluster_name/{clusterName}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = ApiDescription.ClusterOpDescription.SCALING_ACTIVITIES_IN_DURATION_IN_MINUTES,
+            produces = MediaType.APPLICATION_JSON, notes = ApiDescription.DistroXClusterNotes.NOTES)
+    List<DistroXAutoscaleScalingActivityResponse> getScalingActivitiesInDurationByClusterName(
+            @PathParam("clusterName") @NotNull String clusterName,
+            @QueryParam("durationInMinutes") @DefaultValue("60") long durationInMinutes,
+            @QueryParam("page") @DefaultValue("0") Integer page,
+            @QueryParam("size") @DefaultValue("100") Integer size);
+
+    /**
+     * Returns Scaling activities in a particular duration.
+     * @param clusterCrn Crn of cluster for which we require the scaling activities.
+     * @param durationInMinutes the duration in which we want to get all the scaling activities.
+     * @return List of scaling activities after the input time stamp value.
+     */
+    @GET
+    @Path("cluster_crn/{clusterCrn}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = ApiDescription.ClusterOpDescription.SCALING_ACTIVITIES_IN_DURATION_IN_MINUTES,
+            produces = MediaType.APPLICATION_JSON, notes = ApiDescription.DistroXClusterNotes.NOTES)
+    List<DistroXAutoscaleScalingActivityResponse> getScalingActivitiesInDurationByClusterCrn(
+            @PathParam("clusterCrn") @NotNull String clusterCrn,
+            @QueryParam("durationInMinutes") @DefaultValue("60") long durationInMinutes,
+            @QueryParam("page") @DefaultValue("0") Integer page,
+            @QueryParam("size") @DefaultValue("100") Integer size);
+
+    /**
+     * Returns Scaling activity for a particular activity crn value.
+     * @param activityCrn the value for which we want the scaling activity value.
+     * @return A particular scaling activity value associated with the input crn value.
+     */
+    @GET
+    @Path("activity_crn/{activityCrn}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = ApiDescription.ClusterOpDescription.SCALING_ACTIVITY_CRN,
+            produces = MediaType.APPLICATION_JSON, notes = ApiDescription.DistroXClusterNotes.NOTES)
+    DistroXAutoscaleScalingActivityResponse getScalingActivityUsingActivityCrn(@PathParam("activityCrn") @NotNull String activityCrn);
```

Comment:
usually crn is belonging to a resource, not an activity. crn = cloudera resource name

Reply:
also Crn class should be extended here and in thunderhead codebase too. but I would identify any activity based on related cluster or resource crn

Reply:
Here the Scaling Activity is the resource and The activityCrn is the Crn value for it.